spinalcordtoolbox/scripts/sct_deepseg_sc.py

```diff
@@ -184,10 +185,15 @@ def main(argv: Sequence[str]):
 
     im_image = Image(fname_image)
     # note: below we pass im_image.copy() otherwise the field absolutepath becomes None after execution of this function
-    im_seg, im_image_RPI_upsamp, im_seg_RPI_upsamp = \
-        deep_segmentation_spinalcord(im_image.copy(), contrast_type, ctr_algo=ctr_algo,
-                                     ctr_file=manual_centerline_fname, brain_bool=brain_bool, kernel_size=kernel_size,
-                                     threshold_seg=threshold, remove_temp_files=remove_temp_files, verbose=verbose)
+    try:
+        im_seg, im_image_RPI_upsamp, im_seg_RPI_upsamp = \
+            deep_segmentation_spinalcord(im_image.copy(), contrast_type, ctr_algo=ctr_algo,
+                                         ctr_file=manual_centerline_fname, brain_bool=brain_bool,
+                                         kernel_size=kernel_size, threshold_seg=threshold,
+                                         remove_temp_files=remove_temp_files, verbose=verbose)
+    except EmptyArrayError:
+        printv(f"Spinal cord could not be detected (empty array) for {fname_image}", 1, 'error')
+        sys.exit(1)
```

Comment:
@mguaypaq I've followed your suggestions re: catching errors to eliminate the traceback.

However, I'm realizing that catching the errors eliminates some of the useful advice I provided in the original exceptions. So, I was wondering if perhaps we could amend the pattern we're adopting so that we also print the original error message (without reraising). For example:

```suggestion
    except EmptyArrayError:
        printv(f"Spinal cord could not be detected for {fname_image}", 1, 'error')
        printv(f"{e.__class__.__name__}: '{e}'", 1, 'error')
        sys.exit(1)
```

I tried this out, but then realized that `printv` + `'error'` contains its own `sys.exit()` call:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/61ed4a71ceca1ad43b1ccffd76ca8c24ec90a1ad/spinalcordtoolbox/utils/sys.py#L388-L389

This both makes the second `sys.exit(1)` call redundant, and also makes it impossible to print more text after the first `printv` call. This is easily fixed, of course:

```suggestion
    except EmptyArrayError:
        printv(f"Spinal cord could not be detected for {fname_image}\n"
               f"    {e.__class__.__name__}: '{e}'", 1, 'error')
```

But I wanted to mention this because I think it affects the previous caught error that was added in #4200.